Makefile.Common

```diff
@@ -76,7 +76,7 @@ GCI                 := $(TOOLS_BIN_DIR)/gci
 GOTESTSUM           := $(TOOLS_BIN_DIR)/gotestsum
 TESTIFYLINT         := $(TOOLS_BIN_DIR)/testifylint
 
-GOTESTSUM_OPT?= --rerun-fails=1
+GOTESTSUM_OPT?=
```

Comment:
Why is this option removed?

Reply:
enabling the option led to re-running the failed tests (which mostly failed to to go routine leaks and race conditions), which always led to a passed test. This behavior was from some reason blocking the actual problems which were shown after it was removed

Reply:
I added this in #31253 because we have a problem with flaky tests and not enough resources to deal with them. I feel like this is still the case and that if we remove this we will get a more flaky CI which is detrimental to contributors. Is there any way we can detect leaks while still re-running other unit tests?

Reply:
I am not aware of any possibility here. In fact it seems very strange to me that all of the failing tests become passed after re-running them for the 2. time.

We can leave the re-run option in place until we find the root cause why the re-executed tests have very deterministic behavior in terms of go-leaks and data-races, but we should at least consider merging the fixed tests as part of this PR.

Maybe it can be a caching issue in the gotestsum wrapper, not sure here...

Reply:
reported [here](https://github.com/gotestyourself/gotestsum/issues/442)

But there seems to be a bigger problem with the `--rerun-fails` option, as I see multiple issues created on the upstream related to it https://github.com/gotestyourself/gotestsum/issues

Reply:
cc @mx-psi

Reply:
I subscribed to gotestyourself/gotestsum/issues/442 and I would like to wait a bit more to see if upstream has anything to say here. I see some issues related to `--rerun-fails`, but I don't see any specifically related to this.

I am happy to merge the rest of the changes and only keep `--rerun-fails` to a separate PR, would that work for you?

Reply:
ok, reverted the `--rerun-fails` and will keep an eye on the issue opened on gotestsum upstream.

When it will be resolved, will open a new PR